docker/gcp_pudl_etl.sh

```diff
@@ -68,9 +68,13 @@ function copy_outputs_to_gcs() {
 }
 
 function copy_outputs_to_distribution_bucket() {
+    echo "Removing old outputs from GCP distributon bucket."
+    gsutil -m -u $GCP_BILLING_PROJECT rm -r "gs://pudl.catalyst.coop/$GITHUB_REF"
     echo "Copying outputs to GCP distribution bucket"
     gsutil -m -u $GCP_BILLING_PROJECT cp -r "$PUDL_OUTPUT/*" "gs://pudl.catalyst.coop/$GITHUB_REF"
 
+    echo "Removing old outputs from AWS distributon bucket."
+    aws s3 rm "s3://pudl.catalyst.coop/$GITHUB_REF" --recursive
```

Comment:
We've had a couple of instances where we've removed a file, or changed its name, and the old version hangs around because we're relying on new files overwriting old ones to get rid of the old versions.  These explicit purges of all previously distributed files from the `dev` or soon `nightly` paths should avoid that problem.

Reply:
On second thought, this is actually a dangerous operation. In the event that `$GITHUB_REF` is unset, it will wipe out the entire bucket, so I am inclined to hard-code `dev` here and update it to `nightly`/`stable` when we switch over to using those.

Reply:
Okay, now I'm checking that `$GITHUB_REF` is a nonzero length string at least.